### PR TITLE
Don't encode responses with Cache-Control: no-transform

### PR DESCRIPTION
**Pull request by glenjamin**
As discussed in #51 and https://github.com/glenjamin/webpack-hot-middleware/issues/10

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -135,6 +135,14 @@ function compression(options) {
         return
       }
 
+      // Don't compress for Cache-Control: no-transform
+      // https://tools.ietf.org/html/rfc7234#section-5.2.1.6
+      var cacheControl = res.getHeader('Cache-Control') || ''
```
Comment (dougwilson) [inline index.js]: Does this regular expression incorrectly match "foo-no-transform"?

Comment (glenjamin) [inline index.js]: Hrm, you're right - I was specifically trying to catch that case!

I'll add some negative tests and improve this - probably split on `,`, `.trim()` and string compare.

Comment (glenjamin) [inline index.js]: I did have a look around for an existing lib, but didn't spot anything relevant.

Comment (dougwilson) [inline index.js]: Yes, there is no existing lib I know of. I was going to write one and put it in the jshttp lib and use it for the implementation of the feature request, though.

Comment (dougwilson) [inline index.js]: Well, I mean finish the one I started months ago, not start a new one from scratch 😊

**Comment by glenjamin** [comment]
Not the tidiest implementation ever, but should be much more robust than the previous. Thoughts?

**Comment by dougwilson** [comment]
Hi @glenjamin , sorry, I'm not ignoring you :) Was just swamped today, I'm sure you know how it goes :) Yes, seems good enough for now to accept and eventually swap out with a module sometime in the future (at least we'll have the feature today, which is what matter most ;) . I'll get this merged tonight and then release very soon!

**Comment by dougwilson** [comment]
As a side note, I've been debating with myself if we should add an option to disable this feature, but IDK, haha. We don't have a switch to turn off Vary.

I also debated with if this detection should be within the `shouldCompress` function or not (thus allowing people to disable it), but eh :)

This comment was just internal dialog to give to you since you made the PR; not anything that is required to be acted upon :)

**Comment by glenjamin** [comment]
I did wonder about making it configurable or putting into shouldCompress, but came to the same conclusion - if someone's set `no-transform`, we should never transform.

**Comment by glenjamin** [comment]
Sorry to bump, but is this good to merge? Let me know if some changes are still needed.

**Comment by dougwilson** [comment]
Hey! Sorry, it is mostly OK, I was going to merge as part of getting this supported with Node.js 4. It also seems to take noticeable time to run this new code, with the split, the some, the trim, and creating a closure. This can easily be reduced to a single regular expression match, but I have not gotten to it, so if you can do that, awesome, otherwise I'll get to it :) so I technically am not waiting on you, haha.

**Comment by glenjamin** [comment]
Updated with a regex, do you have a benchmark suite or something to measure? I could probably poke about and see if there's a faster regex variant.

**Comment by dougwilson** [comment]
Awesome, @glenjamin , that's it! I should have told you earlier, I'm sorry! As a maintainer, it's always hard to communicate well with people I have not had a history with yet :) Many times, people will just be turned off when there is continued problem noted, so I typically say the first couple, and then adjust after that for a positive experience

**Comment by glenjamin** [comment]
No worries, I've been in your shoes a bunch of times :)

**Comment by dougwilson** [comment]
I'm just getting some dependency updates in here with your change, just as an update.

**Comment by dougwilson** [comment]
Any reasoning for choosing to place the check above the `Vary` response header vs. below it?

**Comment by dougwilson** [comment]
Published to npm as `compression@1.6.0`

**Comment by glenjamin** [comment]
RE: Vary - not sure if I thought about it at the time, but I guess since there's no transforming going on so nothing varies?

**Pull request closed without merging**